experimental/integrations/prototype-spark/marquez-spark-agent/src/main/java/marquez/spark/agent/lifecycle/LogicalPlanTraverser.java

```diff
@@ -0,0 +1,170 @@
+package marquez.spark.agent.lifecycle;
+
+import static scala.collection.JavaConversions.asJavaCollection;
+
+import java.util.Collection;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.sql.SaveMode;
+import org.apache.spark.sql.catalyst.TableIdentifier;
+import org.apache.spark.sql.catalyst.catalog.CatalogStorageFormat;
+import org.apache.spark.sql.catalyst.catalog.CatalogTable;
+import org.apache.spark.sql.catalyst.catalog.CatalogTableType;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.command.CreateDataSourceTableAsSelectCommand;
+import org.apache.spark.sql.execution.datasources.FileIndex;
+import org.apache.spark.sql.execution.datasources.HadoopFsRelation;
+import org.apache.spark.sql.execution.datasources.InsertIntoHadoopFsRelationCommand;
+import org.apache.spark.sql.execution.datasources.LogicalRelation;
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions;
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCRelation;
+import org.apache.spark.sql.sources.BaseRelation;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+public abstract class LogicalPlanTraverser {
```

Comment:
The default behavior is to return  `null` for methods that have not been overridden by a subclass?

Reply:
Yeah, no reason to collect and merge children during traversal. A most robust pattern can be used in the future but most of the time we just want to do an action when a particular node is encountered.

Reply:
Makes sens . Thanks for the context 👍